providers/implementations/encode_decode/encode_key2any.c

```diff
@@ -740,7 +740,12 @@ static int ec_pki_priv_to_der(const void *veckey, unsigned char **pder)
 # define ec_pem_type            "EC"
 
 # ifndef OPENSSL_NO_SM2
-#  define sm2_evp_type          EVP_PKEY_SM2
+/*
+ * SM2 is really an ECC key, and where it matters for encoding (in all places
+ * where an AlgorithmIdentifier is produced, such as PrivateKeyInfo and
+ * SubjectPublicKeyInfo), that's the algorithm OID that should be used.
```

Comment:
What!!!??? This doesn't seem like a good idea. I thought we deliberately moved away from that.

Reply:
Moved away from ....  what?  (not sure what "that" refers to)

Reply:
It's possible that #22184 sheds some light on whatever you're confused about---

Reply:
Deliberately moved away from using the ECC OID for SM2. The ECC OID is for ECDSA which SM2 is not. We used to use the ECC OID for this but I believe we deliberately changed it to use the SM2 OID.